integration/ksm/ksm_test.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -e
+
+dir_path=$(dirname "$0")
+source "${dir_path}/../../lib/common.bash"
+source "${dir_path}/../../metrics/lib/common.bash"
+
+TEST_NAME="Run containers with KSM"
+KATA_KSM_THROTTLER="${KATA_KSM_THROTTLER:-yes}"
+RUNTIME="${RUNTIME:-kata-runtime}"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+IMAGE="${IMAGE:-busybox}"
+WAIT_TIME="60"
+
+function turn_on_ksm() {
+	echo "Turning on KSM"
+	echo "1" | sudo tee "${KSM_ENABLE_FILE}"
+}
+
+function setup() {
+	clean_env
+
+	check_processes
+
+	turn_on_ksm
+
+	check_for_ksm
+}
+
+function teardown() {
+	clean_env
+
+	check_processes
+
+	save_ksm_settings
+
+	restore_ksm_settings
+
+	disable_ksm
+}
```

Comment:
@grahamwhaley , I am using them

Reply:
Ah, I see now. I think we can still use them slightly better:
- use [set_ksm_aggresive](https://github.com/kata-containers/tests/blob/master/metrics/lib/common.bash#L241) to enable KSM, - as it will set up KSM so we 'settle' in a reasonable time.
- and `restore_ksm_settings` will put the KSM enable back to how it was originally.

So, you should probably not (have to, need to, want to) call `enable_ksm` or `disable_ksm` directly, but use `aggresive` and `restore` instead.

I know, the ksm funcs in the lib are probably not well documented on how to use - as they were previously only called from the metrics script I referenced earlier.

Reply:
@grahamwhaley , I will leave the `disable_ksm` to put `0` at `/sys/kernel/mm/ksm/run`, because if I do not call it , I will have
```
restoring KSM settings
$ cat /sys/kernel/mm/ksm/run
1
```

Reply:
Only if:
- KSM was enabled before you called `save_ksm_settings`
- you enabled KSM before calling `save_ksm_settings`.

I'd expect your calls to the ksm library funcs (via a number of your funcs) to go like...

- save ksm settings
- set ksm aggresive mode (which will enable ksm)
- *do work here*
- wait ksm settle
- restore ksm (which will only disable KSM if it was disabled before you called the matching save ksm... thus, trying to leave the system under test as you found it....)

Reply:
@grahamwhaley 👍 thanks, changes applied, sorry that I got confused